gadget/gadget.go

```diff
@@ -69,6 +70,14 @@ type Info struct {
 	Connections []Connection `yaml:"connections"`
 }
 
+// ModelConstraints defines rules to be followed when reading the gadget metadata.
+type ModelConstraints struct {
+	// Classic rules (i.e. content/presence of gadget.yaml is fully optional)
+	Classic bool
+	// System seeding is enabled (Core 20)
```

Comment:
// A system-seed partition (aka recovery partion) is expected (Core 20)

Reply:
Updated comment.